config-service-impl/src/main/java/org/hypertrace/config/service/ConfigServiceGrpcImpl.java

```diff
@@ -125,16 +125,15 @@ public void deleteConfig(
     try {
       ConfigResourceContext configResourceContext = getConfigResourceContext(request);
       ContextSpecificConfig configToDelete = configStore.getConfig(configResourceContext);
-
       // if configToDelete is null/empty (i.e. config value doesn't exist or is already deleted),
       // then throw NOT_FOUND exception
-      if (ConfigServiceUtils.isNull(configToDelete.getConfig())) {
+      if (configToDelete == null || ConfigServiceUtils.isNull(configToDelete.getConfig())) {
```

Comment:
configStore.getConfig has historically been a non-nullable response and it doesn't look like that's changed. If you want to change it, make it return an optional instead.